### PR TITLE
schedulers/docker: Add support for setting environment variables

### PR DESCRIPTION
**Pull request by asuta274**
This change adds support for ``Dict[str, str]`` type in ``runopts`` and use it to expose env flag in ``DockerScheduler``.
The use-case is for setting environment variables for ``local_docker`` scheduler in order to access some remote resources.

Test plan:

```
pytest torchx/schedulers/test/docker_scheduler_test.py
pytest torchx/specs/test/api_test.py
pytest torchx/runner/test/config_test.py
```

### CODE REVIEW

File: torchx/runner/test/config_test.py [hunk 1]
```diff
@@ -105,6 +105,18 @@ def _run_opts(self) -> runopts:
...
+            "d",
+            type_=Dict[str, str],
+            default={"foo": "bar"},
+            help="a dict option",
+        )
+        opts.add(
+            "d_none",
+            type_=Dict[str, str],
+            default=None,
+            help="a None dict option",
```
Comment (d4l3k) [inline torchx/runner/test/config_test.py:118]: can we add a test that use these opts below to verify parsing is correct?

File: torchx/schedulers/docker_scheduler.py [hunk 2]
```diff
@@ -359,6 +364,12 @@ def _run_opts(self) -> runopts:
             default=None,
             help="list of glob patterns of environment variables to copy if not set in AppDef. Ex: FOO_*",
         )
+        opts.add(
+            "env",
+            type_=Dict[str, str],
+            default=None,
+            help="environment varibles to be passed to the run (e.g. ENV1:v1,ENV2:v2,ENV3:v3)",
```
Comment (d4l3k) [inline torchx/schedulers/docker_scheduler.py:371]: can we document that either , or ; will work?

nit: spelling `variables`

Comment (d4l3k) [reply torchx/schedulers/docker_scheduler.py:371]: may also be good to document that this will apply on top of the copied envs

**Review by d4l3k — APPROVED** [review]
LGTM with some small comments

There's some pyre and lint errors -- do you mind fixing them? Thanks

File: torchx/runner/test/config_test.py (hunk 1 above)
Comment (asuta274) [reply torchx/runner/test/config_test.py:118]: Done

File: torchx/schedulers/docker_scheduler.py (hunk 2 above)
Comment (asuta274) [reply torchx/schedulers/docker_scheduler.py]: Done

**Comment by asuta274** [comment]
> LGTM with some small comments
> 
> There's some pyre and lint errors -- do you mind fixing them? Thanks

Done!

**Pull request merged**